org.osgi.test.common/src/main/java/org/osgi/test/common/install/BundleInstaller.java

```diff
@@ -223,10 +228,9 @@ public InputStream openStream(BundleContext bc) throws IOException {
 			URL jarEmbeddedFileUrl = entryStream
 				.findFirst()
 				.orElseThrow(
-					() -> new FileNotFoundException(String.format("File [%s/%s] not found in Bundle [%s:%s]", path,
+					() -> new IOException(String.format("File [%s/%s] not found in Bundle [%s:%s]", path,
```

Comment:
You can still throw FileNotFoundException here. It is a subclass of IOException. My point was about the method declaration's throws clause and not the actual exception thrown.